src/main.cpp

```diff
@@ -5310,12 +5312,16 @@ uint32_t MaxLegacySigops(uint32_t nBlockTime)
     return std::numeric_limits<uint32_t>::max(); // Use accurately-counted limit
 }
 
-uint32_t ForkBits(uint32_t nTime) {
+uint32_t ForkBits(uint32_t nTime, const Consensus::Params& consensusParams) {
     uint32_t bits = 0;
     AssertLockHeld(cs_main);
     // Vote for 2 MB until the fork time to show continued support
-    if (sizeForkTime.load() < nTime && (GetBoolArg("-vote2mb", DEFAULT_2MB_VOTE) || (pblocktree->ForkBitActivated(FORK_BIT_2MB) != uint256())))
+    if (nTime < sizeForkTime.load() &&
+        nTime < consensusParams.SizeForkExpiration() &&
+        (GetBoolArg("-vote2mb", DEFAULT_2MB_VOTE) || (pblocktree->ForkBitActivated(FORK_BIT_2MB) != uint256())))
```

Comment:
It would be nice to move this from main.cpp/main.h into miner.cpp, and not require the cs_main lock.

Can this be simplified to just:

If -vote2MB and time is before fork expiration time:  Vote for 2MB.

Any harm besides reserving the 0x10000000 bit until 2018 in doing that? I believe that bit is unusable until then in any case...